bindings/python/py_src/safetensors/__init__.py

```diff
@@ -1,4 +1,20 @@
 __version__ = "0.2.9"
 
 # Re-export this
-from .safetensors_rust import safe_open  # noqa: F401
+from ._safetensors_rust import safe_open as rust_open, serialize, serialize_file, deserialize, SafetensorError
+
+
+class safe_open:
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+
+    def __getattr__(self, __name: str):
+        return getattr(self.f, __name)
+
+    def __enter__(self):
+        self.f = rust_open(*self.args, **self.kwargs)
+        return self
```

Comment:
Why not just return `self.f` and remove the `__getattr__`?

Reply:
IIUC because you don't want to `del self` in the `__exit__` rather just the return value of `rust_open` given `self` references the `safe_open` ctx manager

Reply:
I'm not sure I understand, it's fine to delete `self.f` in the `__exit__`, all I was suggesting was:
```suggestion
        return self.f
```

Reply:
Because then I wouldn't be able to `del` it. and it would keep things open still.